exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -270,71 +387,71 @@ func (e *Exchange) UpdateTicker(ctx context.Context, currencyPair currency.Pair,
 	if err := e.UpdateTickers(ctx, a); err != nil {
 		return nil, err
 	}
-	return ticker.GetTicker(e.Name, currencyPair, a)
+	pFormat, err := e.GetPairFormat(a, false)
+	if err != nil {
+		return nil, err
+	}
+	return ticker.GetTicker(e.Name, currencyPair.Format(pFormat), a)
 }
 
 // UpdateOrderbook updates and returns the orderbook for a currency pair
 func (e *Exchange) UpdateOrderbook(ctx context.Context, pair currency.Pair, assetType asset.Item) (*orderbook.Book, error) {
 	if pair.IsEmpty() {
 		return nil, currency.ErrCurrencyPairEmpty
 	}
-	if err := e.CurrencyPairs.IsAssetEnabled(assetType); err != nil {
+	err := e.CurrencyPairs.IsAssetEnabled(assetType)
+	if err != nil {
+		return nil, err
+	}
+	pair, err = e.FormatExchangeCurrency(pair, assetType)
+	if err != nil {
 		return nil, err
 	}
-	callingBook := &orderbook.Book{
+	book := &orderbook.Book{
 		Exchange:          e.Name,
 		Pair:              pair,
 		Asset:             assetType,
 		ValidateOrderbook: e.ValidateOrderbook,
 	}
-	orderbookNew, err := e.GetOrderbook(ctx, "", poloniexMaxOrderbookDepth)
-	if err != nil {
-		return callingBook, err
-	}
-
-	enabledPairs, err := e.GetEnabledPairs(assetType)
-	if err != nil {
-		return callingBook, err
-	}
-	for i := range enabledPairs {
-		pFmt, err := e.GetPairFormat(assetType, true)
+	switch assetType {
+	case asset.Spot:
+		var orderbookNew *OrderbookData
+		orderbookNew, err = e.GetOrderbook(ctx, pair, 0, 0)
 		if err != nil {
 			return nil, err
 		}
-		fP := enabledPairs[i].Format(pFmt)
-		data, ok := orderbookNew.Data[fP.Base.String()+fP.Delimiter+fP.Quote.String()]
-		if !ok {
-			data, ok = orderbookNew.Data[fP.Quote.String()+fP.Delimiter+fP.Base.String()]
-			if !ok {
-				continue
-			}
+		book.Bids = make(orderbook.Levels, len(orderbookNew.Bids)/2)
+		for y := range book.Bids {
+			book.Bids[y].Price = orderbookNew.Bids[y*2].Float64()
+			book.Bids[y].Amount = orderbookNew.Bids[y*2+1].Float64()
```

Comment:
This looks cooked. 0 will always be 0 then 1 would be 2 and 3 so it looks like its going to skip a bunch of stuff can you confirm?